pkg/cmd/util/terminal.go

```diff
@@ -37,18 +40,50 @@ func PromptForPasswordString(r io.Reader, format string, a ...interface{}) strin
 			fmt.Printf("\n")
 
 			return input
-		} else {
-			glog.V(3).Infof("Stdin is not a terminal")
-			return PromptForString(r, format, a...)
 		}
-	} else {
-		glog.V(3).Infof("Unable to use a TTY")
+		glog.V(3).Infof("Stdin is not a terminal")
 		return PromptForString(r, format, a...)
 	}
+	return PromptForString(r, format, a...)
+}
+
+// Prompt for user input of a boolean value. The accepted values are:
+//   yes, y, true, 	t, 1 (not case sensitive)
+//   no, 	n, false, f, 0 (not case sensitive)
+// A valid answer is mandatory so it will keep asking until an answer is provided.
+func PromptForBool(r io.Reader, format string, a ...interface{}) bool {
+	str := PromptForString(r, format, a...)
+	switch strings.ToLower(str) {
+	case "1", "t", "true", "y", "yes":
+		return true
+	case "0", "f", "false", "n", "no":
+		return false
+	}
+	fmt.Println("Please enter 'yes' or 'no'.")
+	return PromptForBool(r, format, a...)
+}
+
+// Prompt for user input but take a default in case nothing is provided.
+func PromptForStringWithDefault(r io.Reader, def string, format string, a ...interface{}) string {
```

Comment:
Do we envision any scenarios where we'd want to prompt for a string with a default, where we'd like to allow an empty answer?  Something like clearing an environment variable that had a default value?  If so, add a TODO to come back and fix it.